File: go/backend-validator/07-multiple-subdivisions.go
package main

import (
	"context"
	"encoding/json"
	"fmt"
	"log"

	"github.com/countriesdb/validator-go"
)

func main() {
	// Set your CountriesDB private API key here
	apiKey := "YOUR_API_KEY" // Replace with your actual API key
	if apiKey == "YOUR_API_KEY" {
		log.Fatal("Please set your API_KEY in the script")
	}

	v, err := validator.NewValidator(apiKey)
	if err != nil {
		log.Fatal(err)
	}

	ctx := context.Background()

	// 7. Multiple subdivision validation
	codes := []string{"US-CA", "US-NY", "US-TX"}
	fmt.Printf("Validating subdivision codes: %v (country: US)\n", codes)
	results, err := v.ValidateSubdivisions(ctx, codes, "US", validator.SubdivisionOptions{
		AllowParentSelection: false,
	})
	if err != nil {
		log.Fatalf("Validation failed: %v", err)
	}

	jsonResult, _ := json.MarshalIndent(results, "", "  ")
	fmt.Println("Response:", string(jsonResult))

	for _, result := range results {
		if !result.Valid {
			log.Fatalf("❌ Subdivision %s validation failed: %s", result.Code, result.Message)
		}
	}

	fmt.Println("✅ All subdivisions are valid")
}

File: ruby/backend-validator/07-multiple-subdivisions.rb
require 'bundler/setup'
require 'countriesdb-validator'
require 'json'

# Set your CountriesDB private API key here
api_key = 'YOUR_API_KEY' # Replace with your actual API key
if api_key == 'YOUR_API_KEY'
  abort 'Please set your API_KEY in the script'
end

validator = CountriesDB::Validator.new(api_key: api_key)

# 7. Multiple subdivision validation
codes = ['US-CA', 'US-NY', 'US-TX']
puts "Validating subdivision codes: #{codes.join(', ')} (country: US)"
results = validator.validate_subdivisions(codes, 'US', follow_related: false, allow_parent_selection: false)

puts "Response: #{JSON.pretty_generate(results)}"

results.each do |result|
  unless result[:valid]
    abort "❌ Subdivision #{result[:code]} validation failed: #{result[:message]}"
  end
end

puts '✅ All subdivisions are valid'

